transformer_lens/HookedTransformer.py

```diff
@@ -664,9 +696,26 @@ def load_and_process_state_dict(
                 matrices (W_Q & W_K, and W_O & W_V) to be "even". Defaults to False
             move_state_dict_to_device (bool, optional): Whether to move the state dict to the device of the model. Defaults to True.
         """
-
+        assert (
+            self.cfg.n_devices == 1 or move_state_dict_to_device
+        ), "If n_devices > 1, move_state_dict_to_device must be True"
         if move_state_dict_to_device:
-            state_dict = {k: v.to(self.cfg.device) for k, v in state_dict.items()}
+            for k, v in state_dict.items():
+                if (
+                    k.startswith("embed")
+                    or k.startswith("pos_embed")
+                ):
+                    state_dict[k] = v.to(utils.get_device_for_block_index(0, self.cfg))
+                if (
+                    k.startswith("ln_final")
+                    or k.startswith("unembed")
+                ):
+                    state_dict[k] = v.to(
+                        utils.get_device_for_block_index(self.cfg.n_layers - 1, self.cfg)
+                    )
+                if k.startswith("blocks"):
+                    state_dict[k] = v.to(utils.get_device_for_block_index(int(k.split(".")[1]), self.cfg))
+
```

Comment:
One small thing is that this should be `if... else if... else if...` as it illuminates otherwise hard-to-catch bugs where multiple conditional blocks are called (again we should have a linter that catches this sort of thing - so I'll look into that). Also it should have a final else statement that either throws an error or puts it on a default device (tending towards the former).